keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -0,0 +1,1587 @@
+feature that graduated to beta in 1.25, but with a focus on the policy
+enforcement capabilities of validating admission control.
+
+## Motivation
+
+This KEP will lower the infrastructure barrier to enforcing customizable
+policies as well as providing primitives that help the community establish and
+adhere to best practices of both K8s and its extensions.
+
+Currently the way custom policies are enforced are via admission
+webhooks. Admission webhooks are extremely flexible, but have a few drawbacks as
+compared to in-process policy enforcement:
+
+- They require building infrastructure to host the admission webhook.
+- They contribute to latency by requiring another network hop.
+- Due to the extra infrastructure dependencies, webhooks are inherently less
+  reliable than in-process webhooks. This forces cluster operators to choose
+  between failing closed, which reduces the availability of the cluster as a
+  whole and failing open, which limits the efficacy of webhooks for enforcing
+  policy.
+- Webhooks are operationally burdensome for cluster administrators to
+  manage. They must take responsibility for the observability, security and the
+  release/rollout/rollback plans for the webhook.
+
+Taking a view of the K8s ecosystem as a whole, it is clear that there is demand
+for opinionated policy frameworks.
+
+Pod Security Policies provided this for pods, but encountered a number of
+issues. One of which was that it was hard to keep up with community demand for
+more control surfaces, and the delay in delivering these control surfaces due to
+K8s' rollout period.
+
+Pod Security Admission is a similar solution, but does not attempt to duplicate
+the control granularity that PSP provided.
+
+There are numerous in-tree embedded controllers.
+
+The existence of security regimes like the CIS Kubernetes Benchmarks highlight
+the values of standardized controls. Automating their enforcement, where
+possible, will make it easier for users to lock down their clusters.
+
+With the advent of CRDs, and the drive to make the resources they define
+first-class entities, the footprint of Kubernetes extensions is set to grow for
+the foreseeable future. This KEP allows authors of such extensions to provide
+policy primitives similar to PSP or PSA, putting them on equal footing with
+in-tree functionality.
+
+With the reduced infrastructure footprint and demonstrated demand for a
+customizable, built-in mechanism for extensible policy, this KEP fills a
+community need. It is not intended to replace validating admission webhooks
+altogether, however, since these can support functionality that may not make
+sense to provide in-tree.
+
+### Goals
+
+- Provide an alternative to webhooks for the vast majority of validating
+  admission use cases.
+- Provide the in-tree extensions needed to build policy frameworks for
+  Kubernetes, again without requiring webhooks for the vast majority of use
```

Comment:
Might be good to clarify what type of use cases are ideal for this in-tree extension vs a webhook